src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/SymbolKey/SymbolKey.MethodSymbolKey.cs

```diff
@@ -60,14 +66,24 @@ public static void Create(IMethodSymbol symbol, SymbolKeyWriter visitor)
 
             public static SymbolKeyResolution Resolve(SymbolKeyReader reader, out string? failureReason)
             {
-                var constructedFrom = reader.ReadSymbolKey(out var constructedFromFailureReason);
+                // Methods are members, and members are never resolved in a way where we have contextual types to guide
+                // how the outer parts of the member may resolve.  We can use contextual typing for the *signature*
+                // portion of the member though.
+                var constructedFrom = reader.ReadSymbolKey(
+                    contextualType: null,
+                    out var constructedFromFailureReason);
+
+                using var typeArguments = reader.ReadSymbolKeyArray<IMethodSymbol, ITypeSymbol>(
+                    contextualSymbol: null,
+                    getContextualType: static (_, _) => null,
+                    out var typeArgumentsFailureReason);
+
                 if (constructedFromFailureReason != null)
                 {
                     failureReason = $"({nameof(ConstructedMethodSymbolKey)} {nameof(constructedFrom)} failed -> {constructedFromFailureReason})";
                     return default;
                 }
 
-                using var typeArguments = reader.ReadSymbolKeyArray<ITypeSymbol>(out var typeArgumentsFailureReason);
```

Comment:
this was incorrect logic before. symbolkeys must read their keys to the end, and only return failures once they've done this. otherwise, they end up reading partway through their own key and returning to the caller, who will have no idea how to resume from that point on.